### PR TITLE
feat: enable querying based on mempool status (only gRPC)

### PR DESCRIPTION
**Pull request by tkxkd0159**
## Description

closes: #683 

To avoid conflict with upstream, make another `createQueryContext` function.

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Comment by tkxkd0159** [comment]
`codecov/patch` said new added lines were not tested but it's not true because they were tested by `IntegrationTestSuite`. Those lines were hit when I executed tests at the local machine.

File: baseapp/abci.go [hunk 1]
```diff
@@ -920,3 +921,19 @@ func splitPath(requestPath string) (path []string) {
...
+func (app *BaseApp) createQueryContextWithCheckState() sdk.Context {
+
+	cacheMS := app.checkState.CacheMultiStore()
+
+	// branch the commit-multistore for safety
+	app.checkStateMtx.RLock()
+	ctx := sdk.NewContext(
+		cacheMS, app.checkState.ctx.BlockHeader(), true, app.logger,
+	).WithMinGasPrices(app.minGasPrices).WithBlockHeight(app.LastBlockHeight())
+	app.checkStateMtx.RUnlock()
```
Comment (jaeseung-bae) [inline baseapp/abci.go:936]: Can you move "app.checkStateMtx.RUnlock()" to right after "app.checkStateMtx.RLock()" with "defer" keyword.
If you move "defer app.checkStateMtx.RUnlock()" to L933, Mutex gets closed even if any exceptional case happens.

Comment (tkxkd0159) [reply baseapp/abci.go:936]: Thank you for your comment but the logic between `RLock` is deterministic. So it's impossible to occur any exceptional cases which makes deadlock.

Comment (jaeseung-bae) [reply baseapp/abci.go:936]: Ok, I understand.

**Comment by jaeseung-bae** [comment]
LGTM

**Review by jaeseung-bae — APPROVED** [review]
LGTM

**Review by loin3 — APPROVED** [review]

**Pull request merged**